README.md

```diff
@@ -1,2 +1,422 @@
+        WindowGroup {
+            // Returns a new instance of a `ContentView`.
+            contentViewInstantiator.instantiate()
+        }
+    }
+
+    public init(contentViewInstantiator: Instantiator<ContentView>) {
+        self.contentViewInstantiator = contentViewInstantiator
+    }
+
+    /// A private property that knows how to instantiate a content view.
+    @Instantiated
+    private let contentViewInstantiator: Instantiator<ContentView>
+}
+```
+
+It is possible to write a `Instantiator` with a type-erased generic by utilizing `@Instantiated`‘s `fulfilledByType` parameter.
+
+#### ForwardingInstantiator
+
+The `ForwardingInstantiator` type is how SafeDI enables instantiating any `@Instantiable` type with a `@Forwarded` property. `ForwardingInstantiator` has two generics. The first generic must match the type of the `@Forwarded` property. The second generic matches the type of the to-be-instantiated instance.
+
+```swift
+@Instantiable
+public struct MyApp: App {
+    public var body: some Scene {
+        WindowGroup {
+            if let user = userService.user {
+                // Returns a new instance of a `LoggedInContentView`.
+                loggedInContentViewInstantiator.instantiate(user)
+            } else {
+                // Returns a new instance of a `LoggedOutContentView`.
+                loggedOutContentViewInstantiator.instantiate()
+            }
+        }
+    }
+
+    public init(loggedInContentViewInstantiator: ForwardingInstantiator<User, LoggedOutContentView>, loggedOutContentViewInstantiator: Instantiator<LoggedOutContentView>, userService: UserService) {
+        self.loggedInContentViewInstantiator = loggedInContentViewInstantiator
+        self.loggedOutContentViewInstantiator = loggedOutContentViewInstantiator
+        self.userService = userService
+    }
+
+    /// A private property that knows how to instantiate a logged-in content view.
+    @Instantiated
+    private let loggedInContentViewInstantiator: ForwardingInstantiator<User, LoggedOutContentView>
+
+    /// A private property that knows how to instantiate a logged-out content view.
+    @Instantiated
+    private let loggedOutContentViewInstantiator: Instantiator<LoggedOutContentView>
+
+    @ObservedObject
+    @Instantiated
+    private var userService: UserService
+}
+```
+
+It is possible to write a `ForwardingInstantiator` with a type-erased second generic by utilizing `@Instantiated`‘s `fulfilledByType` parameter.
+
+### Creating the root of your dependency tree
+
+SafeDI automatically finds the root(s) of your dependency tree, and creates an extension on each root that contains a `public init()` method that instantiates the dependency tree.
+
+An `@Instantiable` type qualifies as the root of a dependency tree if and only if:
+
+1. The type‘s SafeDI-injected properties are all `@Instantiated`
+2. The type is not instantiated by another `@Instantiable` type
+
+### Comparing SafeDI and Manual Injection: Key Differences
```

Comment:
To highlight its strength, a useful method is to compare it with other Dependency Injection (DI) frameworks. Presenting the same key points in a table format, using X and ✔️, can effectively illustrate this comparison.

Reply:
I have a `Comparing SafeDI to other DI libraries` section, but it's pretty light-weight. I like the idea of a table here, but I'm not sure we need to go that hard just yet. This section is more of an adoption guide.